relay-metrics/benches/benches.rs

```diff
@@ -1,135 +1,197 @@
-use std::collections::BTreeMap;
-use std::fmt;
-
 use criterion::{black_box, criterion_group, criterion_main, BatchSize, BenchmarkId, Criterion};
+use rand::distributions::Uniform;
+use rand::Rng;
+use rand_pcg::Pcg32;
 use relay_base_schema::project::ProjectKey;
 use relay_common::time::UnixTimestamp;
 use relay_metrics::{
     aggregator::{Aggregator, AggregatorConfig},
     Bucket, BucketValue, DistributionValue, FiniteF64,
 };
+use std::cell::RefCell;
+use std::collections::BTreeMap;
+use std::fmt;
+use std::ops::Range;
 
-/// Struct representing a testcase for which insert + flush are timed.
-struct MetricInput {
-    num_metrics: usize,
-    num_metric_names: usize,
-    num_project_keys: usize,
-    bucket: Bucket,
+struct NumbersGenerator {
+    min: usize,
+    max: usize,
+    current_value: RefCell<usize>,
 }
 
-impl MetricInput {
-    // Separate from actual metric insertion as we do not want to time this function, its logic and
-    // especially not creation and destruction of a large vector.
-    //
-    // In theory we could also create all vectors upfront instead of having a MetricInput struct,
-    // but that would take a lot of memory. This way we can at least free some RAM between
-    // benchmarks.
-    fn get_buckets(&self) -> Vec<(ProjectKey, Bucket)> {
-        let mut rv = Vec::new();
-
-        for i in 0..self.num_metrics {
-            let key_id = i % self.num_project_keys;
-            let metric_name = format!("c:transactions/foo{}", i % self.num_metric_names);
-            let mut bucket = self.bucket.clone();
-            bucket.name = metric_name.into();
-            let key = ProjectKey::parse(&format!("{key_id:0width$x}", width = 32)).unwrap();
-            rv.push((key, bucket));
+impl NumbersGenerator {
+    fn new(range: Range<usize>) -> Self {
+        Self {
+            min: range.start,
+            max: range.end,
+            current_value: RefCell::new(0),
         }
+    }
+
+    fn next(&self) -> usize {
+        let seed = *self.current_value.borrow() as u128;
+        let mut generator = Pcg32::new((seed >> 64) as u64, seed as u64);
```

Comment:
You create the prng only once in the constructor then keep calling it, no need to change the seed everytime.

Reply:
Does it spit always a different number deterministically if we do so?